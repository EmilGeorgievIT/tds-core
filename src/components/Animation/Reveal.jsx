import React from 'react'
import PropTypes from 'prop-types'
import { Transition } from 'react-transition-group'

const defaultStyle = timeout => ({
  transition: `height ${timeout}ms ease-in-out`,
  height: 0,
  overflow: 'hidden',
})

const transitionStyles = {
  entered: { height: '200px' },
}

// TODO Fix the height! Can't do transition with "height: auto"

const Reveal = ({ children, ...rest }) => (
  <Transition {...rest}>
    {status => (
      <div
        style={{
          ...defaultStyle(rest.timeout),
          ...transitionStyles[status],
        }}
      >
        {children()}
      </div>
    )}
  </Transition>
)
Reveal.propTypes = {
  children: PropTypes.func.isRequired,
}

export default Reveal
