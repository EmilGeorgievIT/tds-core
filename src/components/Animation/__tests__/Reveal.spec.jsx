import React from 'react'
import { render } from 'enzyme'

import Reveal from '../Reveal'

describe('Reveal', () => {
  it('renders when the transition is triggered', () => {
    const reveal = render(
      <Reveal timeout={500} in={true}>
        {() => <div>Content to reveal</div>}
      </Reveal>
    )

    expect(reveal).toMatchSnapshot()
  })

  it('renders when the transition is not triggered', () => {
    const reveal = render(
      <Reveal timeout={500} in={false}>
        {() => <div>Content to reveal</div>}
      </Reveal>
    )

    expect(reveal).toMatchSnapshot()
  })
})
