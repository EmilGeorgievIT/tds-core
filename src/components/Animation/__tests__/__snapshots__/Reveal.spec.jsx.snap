// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reveal renders when the transition is not triggered 1`] = `
<div
  style="transition:height 500ms ease-in-out;height:0;overflow:hidden"
>
  <div>
    Content to reveal
  </div>
</div>
`;

exports[`Reveal renders when the transition is triggered 1`] = `
<div
  style="transition:height 500ms ease-in-out;height:200px;overflow:hidden"
>
  <div>
    Content to reveal
  </div>
</div>
`;
